sklearn/cluster/hierarchical.py

```diff
@@ -783,6 +765,10 @@ def fit(self, X, y=None):
                                            n_clusters=n_clusters,
                                            return_distance=True,
                                            **kwargs)
+            # find number of clusters to cut the tree when distance threshold
+            # is reached
+            self.n_clusters = np.count_nonzero(
```

Comment:
But we don't allow modifying constructor parameters during fit (though this is not yet in common tests). So either use a local variable, or, indeed, store the computed n_clusters in a new n_clusters_ attribute (which should then be documented)

Reply:
I've put comments in the code, is there any other place it needs to be documented?